src/pfe/file-watcher/server/test/functional-test/suites/generic/tests/workspace.ts

```diff
@@ -0,0 +1,299 @@
+                    expect(info.msg).to.equal(`The workspace settings file was not found at location: ${settingsPath}`);
+                });
+            });
+
+            describe("read workspace settings with valid contents in file", () => {
+                before("create a backup for original settings file and create a fake settings file", async () => {
+                    if (await fs.existsSync(settingsPath)) {
+                        await fs.copyFileSync(settingsPath, backupSettingsPath);
+                        await fs.writeFileSync(settingsPath, JSON.stringify(settingsContent));
+                    } else {
+                        await fs.writeFileSync(settingsPath, JSON.stringify(settingsContent));
+                        await fs.chownSync(settingsPath, 1001, 2004);
+                    }
+                });
+
+                it("read workspace settings with valid contents ", async () => {
+                    const info: any = await genericLib.readWorkspaceSettings();
+                    expect(info);
+                    expect(info.statusCode);
+                    expect(info.statusCode).to.equal(200);
+                    expect(info.workspaceSettings);
+                    expect(info.workspaceSettings).to.deep.equal(settingsContent);
+                    expect(Object.keys(info.workspaceSettings)).to.deep.equal(Object.keys(settingsContent));
+                    expect(info.workspaceSettings.deploymentRegistry).to.equal(settingsContent.deploymentRegistry);
+                    expect(info.workspaceSettings.watcherChunkTimeout).to.equal(settingsContent.watcherChunkTimeout);
+                });
+            });
+
+            describe("read workspace settings with invalid deployment registry in file", () => {
+                const badSettingsContent = _.cloneDeep(settingsContent);
+                badSettingsContent.deploymentRegistry = "x^72@!$&";
+
+                before("create a backup for original settings file and create a fake settings file", async () => {
+                    if (await fs.existsSync(settingsPath)) {
+                        await fs.copyFileSync(settingsPath, backupSettingsPath);
+                        await fs.writeFileSync(settingsPath, JSON.stringify(badSettingsContent));
+                    } else {
+                        await fs.writeFileSync(settingsPath, JSON.stringify(badSettingsContent));
+                        await fs.chownSync(settingsPath, 1001, 2004);
+                    }
+                });
+
+                it("read workspace settings with invalid deployment registry", async () => {
+                    const info: any = await genericLib.readWorkspaceSettings();
+                    const failureMsg = "Codewind detected an error with the Deployment Registry x^72@!$&. Please ensure it is a valid Deployment Registry.";
+                    const targetEvent = eventConfigs.events.deploymentRegistryStatus;
+                    console.log("Info: %j", info);
```

Comment:
Is this `console.log()` a debug code for test cases? We can remove this, otherwise it's pretty confused that it's from PFE log or from Mocha tests

![Screen Shot 2019-08-12 at 2 43 52 PM](https://user-images.githubusercontent.com/15131537/62889542-a0b6bb80-bd0f-11e9-8659-144529fa3ce5.png)

Reply:
Yeah that was for test debugging. Will remove it.